Configuration/TCA/Overrides/sys_file_reference.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+use TYPO3\CMS\Core\Utility\ExtensionManagementUtility;
+
+$displayCondition = [
+    'AND' => [
+        'FIELD:tablenames:=:tx_html5mediakit_domain_model_media',
+        'FIELD:fieldname:=:tracks',
+    ],
+];
+
+$additionalColumns = [
+    'tx_html5mediakit_track_kind' => [
+        'label' => 'LLL:EXT:html5mediakit/Resources/Private/Language/locallang_db:tx_html5mediakit_track_kind',
+        'config' => [
+            'type' => 'select',
+            'renderType' => 'selectSingle',
+            'items' => [
+                ['LLL:EXT:html5mediakit/Resources/Private/Language/locallang_db:tx_html5mediakit_track_kind', '--div--'],
+                ['subtitles', 'subtitles'],
+                ['captions', 'captions'],
+                ['descriptions', 'descriptions'],
+                ['chapters', 'chapters'],
+                ['metadata', 'metadata'],
+            ],
+            'default' => 'subtitles',
+        ],
+        'displayCond' => $displayCondition,
+    ],
+    'tx_html5mediakit_track_label' => [
+        'label' => 'LLL:EXT:html5mediakit/Resources/Private/Language/locallang_db:tx_html5mediakit_track_label',
+        'config' => [
+            'type' => 'input',
+            'eval' => 'trim',
+            'size' => 10,
+        ],
+        'displayCond' => $displayCondition,
+    ],
+    'tx_html5mediakit_track_srclang' => [
+        'label' => 'LLL:EXT:html5mediakit/Resources/Private/Language/locallang_db:tx_html5mediakit_track_srclang',
+        'description' => 'LLL:EXT:html5mediakit/Resources/Private/Language/locallang_db:tx_html5mediakit_track_srclang.description',
+        'config' => [
+            'type' => 'input',
+            'eval' => 'trim,required,alpha,nospace,lower,2',
```

Comment:
I think we can not make this field required because according to [the spec](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/track#srclang), the field is optional, if `kind` is not `subtitles`:

> If the kind attribute is set to subtitles, then srclang must be defined.

Another idea would be to use two fields and switch them with a display condition depending on the selected `kind`:

* `tx_html5mediakit_track_srclang_optional`
* `tx_html5mediakit_track_srclang_required`

They would both have the same config except the `displayCond` and  the [`required`](https://docs.typo3.org/m/typo3/reference-tca/main/en-us/ColumnsConfig/CommonProperties/Required.html) config (using `required` in eval [is deprecated](https://docs.typo3.org/m/typo3/reference-tca/main/en-us/ColumnsConfig/Type/Input/Properties/Eval.html)).

Also I must confess I have never seen a number being used in the eval field to restrict the length of the value and I did not find it in the docs. But we can use `min` and `max` instead:

```suggestion
            'eval' => 'trim,alpha,nospace,lower',
            'min' => 2,
            'max' => 2,
```

Reply:
Thank you for your feedback! I appreciate your attention to detail.
Should we leave it as optional for now to avoid more complication?

Reply:
Agreed, let's keep it simple :+1:

Reply:
Then I would say! that's it! we could happly merge this PR.